src/Microsoft.AspNet.TestHost/TestServer.cs

```diff
@@ -129,7 +127,7 @@ public Task Invoke(HttpContext context)
             {
                 throw new ObjectDisposedException(GetType().FullName);
             }
-            return _appDelegate(context);
+            return _requestHandler(context.Features);
```

Comment:
It's strange to take an HttpContext here, pull out the features, and then create a new one.